### ISSUE TITLE
Travis & OpenSSL 1.0.2 & 1.1.0

### ISSUE THREAD

**Issue by MSP-Greg** [eventmachine/eventmachine#790]
While working on Travis CI (I"m a windows guy) for the recent PR for 1.1.0, I noticed that most of the ruby builds seem to be using OpenSSL 1.0.1.  Per the [OpenSSL web site](https://www.openssl.org/source/),  'The 0.9.8, 1.0.0 and 1.0.1 versions are now out of support and should not be used.'

I noticed that the Ruby OpenSSL gem has the following in its [travis.yml](https://github.com/ruby/openssl/blob/master/.travis.yml) file:
```
--
matrix:
  fast_finish: true
  include:
    - env: RUBY_VERSION=ruby-2.3 OPENSSL_VERSION=openssl-1.0.2
    - env: RUBY_VERSION=ruby-2.4 OPENSSL_VERSION=openssl-1.0.1
    - env: RUBY_VERSION=ruby-2.4 OPENSSL_VERSION=openssl-1.0.2
    - env: RUBY_VERSION=ruby-2.4 OPENSSL_VERSION=openssl-1.1.0
```

I'm not familiar with travis scripts, but it sure looks like they're building/testing against 1.0.2 and 1.1.0.  Might be something to look into for EM...

**Comment by sodabrew** [comment]
That's a good find! Thanks for the suggestion.

**Comment by MSP-Greg** [comment]
@sodabrew 

I know the topic says Travis, but it also has the OpenSSL versions.  As mentioned above, I believe all the Ruby versions on Travis are using 1.0.1 or earlier, with the exception of XCode.

Over on my fork, I added an appveyor branch, along with [a commit](https://github.com/MSP-Greg/eventmachine/commit/621fe89e89cadacc4ea7b218ec33f6d2f8424abd).  The [appveyor.yml](https://github.com/MSP-Greg/eventmachine/blob/621fe89e89cadacc4ea7b218ec33f6d2f8424abd/appveyor.yml) file has Ruby 2.4.1 using OpenSSL 1.0.2, and Ruby trunk using 1.1.0.  [Appveyor results](https://ci.appveyor.com/project/MSP-Greg/eventmachine) & [Travis results](https://travis-ci.org/MSP-Greg/eventmachine/builds/259833902).

I tried adding Ruby 2.3 to the matrix, I got a segfault.  I'll try to look at it, but, at present, Ruby 2.3 and earlier use a different and much older build system.

So, if you'd like to add appveyor testing again, I'd be happy to do the PR.

As an aside, I've noticed the 'EM.popen not implemented' message in the tests.  Not sure what that means, but I think std-lib 'open3' works fine with these builds.  Might it work now?

* Info about my trunk build

At present ruby has four test groups.  Previous to my work, I believe that no one has regularly (fully) tested Ruby MinGW builds, trunk or otherwise.  I also run one test suite again from a 'windows cmd shell', rather than the 'combination shell' that the normal four tests run in.

Long story short, the tests are not consistent (I run all parallel).  So, due to the fact that Appveyor has dual cores, I have a quad, and the test issues, I build & check the trunk at home, then push it to [BinTray](https://dl.bintray.com/msp-greg/ruby_windows/), along with build/test logs.  I bring this up because some people feel that if they can't see the build on Appveyor, they won't use it.  I'm not asking anyone to use the build in a sensitive commercial or government app, I'm just making it available for testing, especially because it's one of very few available ruby builds with 1.1.0 (which I also created a custom MinGW package for).

Thanks again...

**Comment by MSP-Greg** [comment]
@sodabrew Okay to close by me...

**Comment by MSP-Greg** [comment]
Travis is currently testing with OpenSSL 1.0.2 & 1.1.0. Future Appveyor will use OpenSSL 1.1.1.  Closing.
